SignInCheckIn/SignInCheckIn/Services/ChildSigninService.cs

```diff
@@ -419,97 +417,90 @@ public void CreateGroupParticipants(string token, List<MpNewParticipantDto> mpPa
+                    {
+                        nextServiceEvent = dailyEvents.First(r => r.EventId != participantEventMapDto.CurrentEvent.EventId && r.ParentEventId == null);
+                    }
 
-                    // sign them into a regular service
-                    if (nextServiceEvent == null || nextServiceEvent.EventId == participantEventMapDto.CurrentEvent.EventId)
+                    // Case #2 - no following service events, sign them into the current service event
+                    if (nextServiceEvent == null)
                     {
-                        returnEventIds.Add(participantEventMapDto.CurrentEvent.EventId);
-                        return returnEventIds;
+                        returnEvents.Add(dailyEvents.First(r => r.EventId == participantEventMapDto.CurrentEvent.EventId));
+                        return returnEvents;
                     }
 
-                    // if the event being signed into is an ac event and there is a later ac event,
-                    // sign them into the current service and the later ac event
-                    if (nextServiceEvent.EventId == participantEventMapDto.CurrentEvent.EventId)
+                    // Case #3 - no AC event for current event, but later AC, sign them into the current
+                    // event and later AC
+                    if (mpAcEventDto.ParentEventId != participantEventMapDto.CurrentEvent.EventId)
                     {
-                        returnEventIds.Add(participantEventMapDto.CurrentEvent.EventId);
-                        returnEventIds.Add(nextAcEvent.EventId);
-                        return returnEventIds;
+                        returnEvents.Add(dailyEvents.First(r => r.EventId == participantEventMapDto.CurrentEvent.EventId));
+                        returnEvents.Add(mpAcEventDto); 
+                        return returnEvents;
                     }
 
-                    // ReSharper disable once ConditionIsAlwaysTrueOrFalse -- JPC
-                    if (nextServiceEvent != null && nextAcEvent.ParentEventId == participantEventMapDto.CurrentEvent.EventId)
+                    // Case #4 - AC for current event and later event exists with no AC, sign them 
+                    // into the current AC event and later service event
+                    if (mpAcEventDto.ParentEventId == participantEventMapDto.CurrentEvent.EventId &&
+                        nextServiceEvent != null)
                     {
-                        returnEventIds.Add(participantEventMapDto.CurrentEvent.EventId);
-                        returnEventIds.Add(nextAcEvent.EventId);
-                        return returnEventIds;
+                        returnEvents.Add(nextServiceEvent);
+                        returnEvents.Add(mpAcEventDto);
+                        return returnEvents;
                     }
-                }
+
+                    // Case #5 - there are AC events for both the current and later service event. For now, pending
```

Comment:
Remove this case as we are currently doing it above.  We will always just sign them into the first event we can check them into AC.